test/e2e/kubectl/portforward.go

```diff
@@ -695,11 +695,11 @@ func wsWrite(conn *websocket.Conn, channel byte, data []byte) error {
 	return err
 }
 
-func post(url string, reader io.Reader, transport *http.Transport) (string, error) {
-	if transport == nil {
-		transport = utilnet.SetTransportDefaults(&http.Transport{})
+func post(url string, reader io.Reader, timeout time.Duration) (string, error) {
+	client := &http.Client{
+		Transport: utilnet.SetTransportDefaults(&http.Transport{}),
+		Timeout:   timeout,
```

Comment:
Hmm… a client side timeout will keep the request from hanging, but is it actually testing what we want? I thought we wanted to make sure the portforward handler itself wouldn't hang

Reply:
It's the current limitation of this and other tests relying on port-forward. I'm planning to rewrite it to have a control over the port-forward invocation, but that unfortunately requires a major re-write that I'm planning to handle in the next weeks.

Reply:
if we actually hit this timeout, we'll get back a non-nil error, but that error will fail this check, right?

`gomega.Expect(err.Error()).To(gomega.Or(gomega.ContainSubstring("connection reset by peer"), gomega.ContainSubstring("EOF")))`

Reply:
I need to run this check locally and try it out, I didn't have the time earlier today. I'll do that tomorrow first thing in the morning and will update accordingly.

Reply:
Did some tests, and added there match for context timeout, which should happen in that case.